test/e2e/spec/dyndns_spec.js

```diff
@@ -0,0 +1,289 @@
+                    token = loginResponse.body.auth_token;
+                    chakram.setRequestHeader('Authorization', 'Token ' + token);
+                });
+            });
+        });
+
+        describe("(logged in)", function () {
+
+            before(function () {
+
+            });
+
+            describe("api 'homepage'", function () {
+
+                var response;
+
+                before(function () {
+                    response = chakram.get('/');
+                });
+
+                it('has status 200', function () {
+                    return expect(response).to.have.status(200);
+                });
+
+                it('looks according to the schema', function () {
+                    return expect(response).to.have.schema(apiHomeSchema);
+                });
+
+            });
+
+            describe("domains endpoint", function () {
+
+                it("can register a domain name", function () {
+                    var domain = 'e2etest-' + require("uuid").v4() + '.dedyn.io';
+                    return expect(chakram.post('/domains/', {'name': domain})).to.have.status(201);
+                });
+
+            });
+
+            describe("a domain endpoint", function () {
+
+                var domain;
+
+                before(function () {
+                    domain = 'e2etest-' + require("uuid").v4() + '.dedyn.io';
+                    return expect(chakram.post('/domains/', {'name': domain})).to.have.status(201);
+                });
+
+                it("can set an IPv4 address", function () {
+                    return expect(chakram.post(
+                        '/domains/' + domain + '/rrsets/',
+                        {
+                            'subname': '',
+                            'type': 'A',
+                            'records': ['127.0.0.1'],
+                            'ttl': 60,
+                        }
+                    )).to.have.status(201);
+                });
+
+                it("can set an IPv6 address", function () {
+                    return expect(chakram.post(
+                        '/domains/' + domain + '/rrsets/',
+                        {
+                            'subname': '',
+                            'type': 'AAAA',
+                            'records': ['::1'],
+                            'ttl': 60,
+                        }
+                    )).to.have.status(201);
+                });
+
+            });
+
+        });
+
+        var domain = 'e2etest-' + require("uuid").v4() + '.dedyn.io';
+        describe("and domain [" + domain + "]", function () {
+
+            before(function () {
+                chakram.setRequestHeader('Authorization', 'Token ' + token);
+                return expect(chakram.post('/domains/', {'name': domain})).to.have.status(201);
+            });
+
+            describe("dyndns12 endpoint with basic auth", function () {
```

Comment:
add tests with empty IPv4 and/or IPv6 argument and test record deletion

Reply:
done:
1) https://github.com/desec-io/desec-stack/pull/71/files#diff-630fb43ba6d02fbc6b856ae12276a939R230
2) https://github.com/desec-io/desec-stack/pull/71/files#diff-630fb43ba6d02fbc6b856ae12276a939R279